drivers/entropy/Kconfig

```diff
@@ -5,11 +5,10 @@
 
 menuconfig ENTROPY_GENERATOR
 	bool "Entropy drivers"
+	default y if RNG || CSRNG
 	help
 	  Include entropy drivers in system config.
 
-if ENTROPY_GENERATOR
```

Comment:
Removing this was necessary due to circular dependency problems

Reply:
When I tried to do something similar in #82497 @nordicjm told me that this was not correct. In my case I was auto-enabling `CONFIG_ENTROPY_GENERATOR` whenever "zephyr,entropy" chosen property was set, while here we are completely removing that Kconfig, but I think the conclusion is similar: there would be no "high level" Kconfig symbol which is gating the platform sub-Kconfigs...
@nordicjm wdyt?

Reply:
There should be a top level Kconfig controlling all sub-configs, you should not see or be able to configure e.g. the log level of a feature if the feature is not enabled so removing this is wrong

Reply:
we say that but the we have a annoying dependency tree between ENTROPY_GENERATOR and enabling random. I'm going to leave this like it is for now while we discuss the larger objective here.

Reply:
The weirdness we have in the system is that IF you want to access the rand/csrand functions the only way to do this is to enable ENTROPY_GENERATOR, which is also the enablement for ENTROPY and entropy_get_entropy(). We also made this overall configuration such that the application needs to know to set this configuration if it happens to include some subsystem/module component that needs rand or entropy. I want to get the subsystems and modules that know they need rand and/or entropy to automatically set the appropriate Kconfig values instead of moving that responsibility to the application layer configuration.

The only thing I want the application layer configuration to have to set is the TEST_RANDOM_GENERATOR to make an overt decision to use something that is not really a valid generator.

Reply:
> The weirdness we have in the system is that IF you want to access the rand/csrand functions the only way to do this is to enable ENTROPY_GENERATOR, which is also the enablement for ENTROPY and entropy_get_entropy().

Let me add to this that enabling `ENTROPY_GENERATOR` is the only way to poll if there is any way to really have some entropy driver enabled for that platform or not. Ideally one would first try to enable that, see what happens, and then decide whether they can use it or need to rely on some fake/test generator.
Of course this is not possible because parsing of Kconfig is done in a single pass, so that's why I came up with `CSPRNG_AVAILABLE` in #82859.

Reply:
It's worth looking at other systems and selections in zephyr that do this, for example the C library selection - there is minimal, picolibc, newlib and external, there is a Kconfig that applications (and systems) can select to say that they **require** a full C library, that is, not the minimal C library:
```
  config REQUIRES_FULL_LIBC
        bool "Require complete C library"
        help
          Select a C library implementation that provides a complete C library
          implementation, rather than the subset provided by MINIMAL_LIBC.
```
This is selected by the following options:
```
SIMPLELINK_HOST_DRIVER
LIBLC3
IEEE802154_KW41Z
NPM1300_CHARGER
REQUIRES_FULL_LIBCPP
POSIX_DEVICE_IO
LORAWAN
UPDATEHUB
```
Then there is an option which specifies if there is a full C library available:
```
  config FULL_LIBC_SUPPORTED
        bool
        help
          Selected when the target has at least one C library that offers a
          complete implementation and which would be selected when
          REQUIRES_FULL_LIBC is set.
```
Which is selected by the following:
```
NEWLIB_LIBC_SUPPORTED
PICOLIBC_SUPPORTED
NATIVE_LIBC
```
Then there is another Kconfig which selects which C library you have active, because whilst it might default to picolibc, perhaps you want to use newlib. That sort of design and implementation should be reused for entropy

Reply:
the circular dependency is the ENTROPY_HAS_DRIVER. Looking at the examples you highlight to see how to work around this.

Reply:
I used the hint @valeriosetti suggested and now have a ENTROPY_AVAILABLE config driven from DT node. This helps break the dependency loops with HAS_DRIVER.